### PR TITLE
AP-2046-G Use ApplicationProceedingTypesScopeLimtiations model

### PR DESCRIPTION
**Pull request by stephenrichards**
## Use ApplicationProceedingTypesScopeLimtiations model

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2046)

This PR has been approved once, but then split into two parts: the actual migration of records from `application_scope_limitations` to `application_proceeding_types_scope_limitations`, and then using the new records as opposed to the old.
The migration has been done, and this PR is to use the new `application_proceeding_types_scope_limitations` records.

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/webpack/stylesheets/interruption-panel.scss [hunk 1]
```diff
@@ -2,8 +2,12 @@
   background-color:#1d70b8;
   padding:30px;
 
-  p,h1,h2,ul,label,a,a:link,a:visited {
```
Comment (mpw5) [inline app/webpack/stylesheets/interruption-panel.scss:5]: I don't think the changes to this file are needed. This is reverting changes that were made for the DWP Override functionalilty and are needed on master.

File: spec/requests/providers/check_benefits_spec.rb [hunk 2]
```diff
@@ -92,29 +92,6 @@
...
-      end
-
-      context 'when the check benefit result is negative' do
-        let(:application) { create :legal_aid_application, :with_applicant, :with_negative_benefit_check_result, :at_checking_applicant_details }
-
-        it 'displays the confirm dwp non passported_applications page' do
-          expect(response).to redirect_to providers_legal_aid_application_confirm_dwp_non_passported_applications_path(application)
-        end
-      end
-    end
```
Comment (mpw5) [inline spec/requests/providers/check_benefits_spec.rb:117]: I don't think the changes to this file are needed. This is reverting changes that were made for the DWP Override functionalilty and are needed on master.

File: spec/requests/providers/check_provider_answers_spec.rb [hunk 3]
```diff
@@ -209,8 +209,8 @@
         end
 
         context 'non passported' do
-          it 'redirects to the check benefits page' do
-            expect(response).to redirect_to(providers_legal_aid_application_check_benefits_path(application))
+          it 'redirects to the confirm dwp non passported applications page' do
+            expect(response).to redirect_to(providers_legal_aid_application_confirm_dwp_non_passported_applications_path(application))
```
Comment (mpw5) [inline spec/requests/providers/check_provider_answers_spec.rb:213]: I don't think the changes to this file are needed. This is reverting changes that were made for the DWP Override functionalilty and are needed on master.

**Review by mpw5 — CHANGES_REQUESTED** [review]
Your changes all look good, but this PR seems to be reverting some other changes previously added for the DWP Override. I think it's probably just a rebase thing. 

The changes to `app/webpack/stylesheets/interruption-panel.scss`, `spec/requests/providers/check_benefits_spec.rb` and `spec/requests/providers/check_provider_answers_spec.rb` need to be removed.

File: app/services/flow/flows/provider_start.rb [hunk 4]
```diff
@@ -61,18 +61,28 @@ class ProviderStart < FlowSteps # rubocop:disable Metrics/ClassLength
...
+            if Setting.override_dwp_results?
+              application.non_passported? ? :confirm_dwp_non_passported_applications : :check_benefits
+            else
+              :check_benefits
+            end
+          end
         },
         check_benefits: {
           path: ->(application) { urls.providers_legal_aid_application_check_benefits_path(application) },
-          forward: ->(application, dwp_override_non_passported) do
```
Comment (willc-work) [inline app/services/flow/flows/provider_start.rb:68]: Is this dwp_override supposed to be removed?

File: features/step_definitions/civil_journey_steps.rb [hunk 5]
```diff
@@ -96,10 +96,6 @@
   Setting.setting.update!(allow_multiple_proceedings: true)
 end
 
-Given('the setting to allow DWP overrides is enabled') do
```
Comment (willc-work) [inline features/step_definitions/civil_journey_steps.rb:99]: dwp_override code

File: app/services/flow/flows/provider_start.rb (hunk 4 above)
Comment (mpw5) [reply app/services/flow/flows/provider_start.rb:68]: Good catch. This ties in with my comments on other files. Obviously I didn't spot all of them 🤦

File: features/step_definitions/civil_journey_steps.rb [hunk 6]
```diff
@@ -426,37 +421,6 @@
   steps %(Then I should be on a page showing 'Check your answers')
 end
 
-Given('I complete the non-passported journey as far as check your answers') do
```
Comment (willc-work) [inline features/step_definitions/civil_journey_steps.rb:429]: dwp_override code

File: spec/factory_specs/legal_aid_application_factory_spec.rb [hunk 7]
```diff
@@ -48,76 +48,238 @@
...
+               this_proceeding_type: pt1,
+               substantive_scope_limitation: sl1
+      end
+      before { laa }
+
+      it 'attaches the subst scope limitation to the proceeding type as a default' do
+        expect(pt1.default_substantive_scope_limitation).to eq sl1
+      end
+
+      it 'does not attach a default delegated fucntions scope limitation to the proceeding type' do
```
Comment (willc-work) [inline spec/factory_specs/legal_aid_application_factory_spec.rb:269]: nitpick - spelling of functions

File: spec/requests/providers/check_benefits_spec.rb [hunk 8]
```diff
@@ -92,29 +92,6 @@
       before { subject }
       it_behaves_like 'a provider not authenticated'
     end
-
```
Comment (willc-work) [inline spec/requests/providers/check_benefits_spec.rb:95]: dwp_override code

**Review by mpw5 — APPROVED** [review]
LGTM

**Review by willc-work — APPROVED** [review]

**Pull request merged**